recipe/patches/0001-add-pthread-ucrt.patch

```diff
@@ -0,0 +1,31 @@
+From 4357f1f031a32f078d9863e53b2a68534e845cf2 Mon Sep 17 00:00:00 2001
+From: Mervin Fansler <mervin.fansler@bric.ku.dk>
+Date: Tue, 22 Oct 2024 10:42:04 +0200
+Subject: [PATCH] add pthread ucrt
+
+---
+ src/Makevars.ucrt | 12 +++---------
+ 1 file changed, 3 insertions(+), 9 deletions(-)
+
+diff --git a/src/Makevars.ucrt b/src/Makevars.ucrt
+index 76d6a15..f3b4bf2 100644
+--- a/src/Makevars.ucrt
++++ b/src/Makevars.ucrt
+@@ -1,14 +1,8 @@
+-
+-ifeq (,$(shell pkg-config --version 2>/dev/null))
+-  PKG_LIBS= \
++PKG_LIBS= \
+ 	-lwinhttp -lcurl -lssh2 -lz -lssl -lcrypto -lgdi32 -lcrypt32 -lwldap32 \
+-        -lzstd -lgcrypt -lgpg-error -lidn2 -lbcrypt -lunistring -liconv -lws2_32
+-else
+-  PKG_LIBS=$(shell pkg-config --libs libcurl)
+-endif
+-
++        -lzstd -lbcrypt -liconv -lws2_32 -pthread
+ PKG_CPPFLAGS= \
+-	-DCURL_STATICLIB -DSTRICT_R_HEADERS
++	-DCURL_STATICLIB -DSTRICT_R_HEADERS -I"$(shell echo $(LIBRARY_INC) | sed 's|\\|/|g')"
```

Comment:
This still doesn't work (note the lack of any slashes in the include path) 😮‍💨 

```cpp
echo '#include <curl/curl.h>' | x86_64-w64-mingw32-gcc  -DCURL_STATICLIB -DSTRICT_R_HEADERS -I"D:bldr-strawr_1729598428420_h_envLibraryinclude" -std=gnu99 -E -xc - | grep "^[ \t]*CURLOPT_.*," | sed s/,// > ../tools/option_table.txt
<stdin>:1:10: fatal error: curl/curl.h: No such file or directory
compilation terminated.
```

But the build passes, so I give up and will merge as is.